weblab/fitting/views.py

```diff
@@ -197,3 +204,150 @@ def get(self, request, *args, **kwargs):
+        protocols = ProtocolEntity.objects.visible_to_user(request.user)
+        protocol_versions = CachedProtocolVersion.objects.visible_to_user(request.user)
+        fittingspecs = FittingSpec.objects.visible_to_user(request.user)
+        fittingspec_versions = CachedFittingSpecVersion.objects.visible_to_user(request.user)
+        datasets = Dataset.objects.visible_to_user(request.user)
+
+        def _get_int_param(fieldname, _model):
+            try:
+                pk = int(request.GET.get(fieldname, ''))
+                return _model.objects.get(pk=pk)
+            except ValueError:
+                pass
+            except ObjectDoesNotExist:
+                pass
+
+        model = _get_int_param('model', ModelEntity)
+        protocol = _get_int_param('protocol', ProtocolEntity)
+        fittingspec = _get_int_param('fittingspec', FittingSpec)
+        dataset = _get_int_param('dataset', Dataset)
+
+        if not protocol:
+            if fittingspec:
+                protocol = fittingspec.protocol
+            elif dataset:
+                protocol = dataset.protocol
+
+        # Restrict to specified model and its versions
+        if model:
+            models = [model]
+            model_versions = model_versions.filter(entity__entity=model)
+
+        # Restrict to specified fitting spec and its versions
+        if fittingspec:
+            fittingspecs = [fittingspec]
+            fittingspec_versions = fittingspec_versions.filter(entity__entity=fittingspec.id)
+
+        # Restrict to specified dataset
+        if dataset:
+            datasets = [dataset]
+
+        # Restrict to specified protocol and its versions
+        if protocol:
+            protocols = [protocol]
+            protocol_versions = protocol_versions.filter(entity__entity=protocol.id)
+
+            # If no fitting spec was chosen yet, restrict to those linked to this protocol
+            if not fittingspec:
+                fittingspecs = fittingspecs.filter(protocol=protocol.id)
+                fsids = [fs.pk for fs in fittingspecs.filter(protocol=protocol.id)]
+                fittingspec_versions = fittingspec_versions.filter(entity__entity__in=fsids)
+
+            # If no dataset was chosen yet, restrict to those linked to this protocol
+            if not dataset:
+                datasets = datasets.filter(protocol=protocol.id)
+
+        # These might be either querysets or ID lists
+        def _get_ids(qs):
+            return [item.id for item in qs]
+
+        options['models'] = _get_ids(models)
+        options['model_versions'] = _get_ids(model_versions)
+        options['protocols'] = _get_ids(protocols)
+        options['protocol_versions'] = _get_ids(protocol_versions)
+        options['fittingspecs'] = _get_ids(fittingspecs)
+        options['fittingspec_versions'] = _get_ids(fittingspec_versions)
+        options['datasets'] = _get_ids(datasets)
+
+        return JsonResponse({
+            'fittingResultOptions': options
```

Comment:
I'm not sure where this info is fed to the user-visible page, but it'd be nice if the drop-downs for entity version displayed a more user-friendly identifier than e.g. "my model@very long sha"! The entity name is redundant since the user has selected an entity just above. And tag names would be preferable to SHAs where they exist, as is done when viewing an entity version.